whylog/config/parser_name_manager.py

```diff
@@ -0,0 +1,48 @@
+import re
+
+from ordered_set import OrderedSet
+
+
+class ParserNameManager(object):
+    def __init__(self, parsers):
+        self._parsers_name = OrderedSet()
+        for parser_name in parsers:
+            self._parsers_name.add(parser_name)
+
+    def add_new_parser_name(self, parser_name):
+        self._parsers_name.add(parser_name)
+
+    def is_free_parser_name(self, parser_name, black_list):
+        return (parser_name not in self._parsers_name) and (parser_name not in black_list)
+
+    def propose_parser_name(self, line, regex_str, black_list):
+        building_words = ParserNameManager._get_building_words(line, regex_str)
+        if not building_words:
+            return self._find_free_by_number_appending('parser_name', black_list)
+        if len(building_words) == 1:
```

Comment:
lines: 22-26 are very similiar to 28-31
maybe replace it with `function(words_count, words_list, black_list)` that returns proposed name. `words_count` here will be 1 and 2, `words_list` will be `building_words`.